core/domain/opportunity_services.py

```diff
@@ -0,0 +1,380 @@
+            # list as exploration can be voiceovered in it's own language.
+            need_voice_artist_in_languages.add(exploration.language_code)
+
+        content_count = exploration.get_content_count()
+        translation_counts = exploration.get_translation_counts()
+
+        exploration_opportunity_summary = (
+            opportunity_domain.ExplorationOpportunitySummary(
+                exp_id, topic.id, topic.name, story.id, story.title, node.title,
+                content_count, list(incomplete_translation_languages),
+                translation_counts, list(need_voice_artist_in_languages), []))
+
+        exploration_opportunity_summary_list.append(
+            exploration_opportunity_summary)
+
+    _save_multi_exploration_opportunity_summary(
+        exploration_opportunity_summary_list)
+
+
+def update_opportunity_with_updated_exploration(exp_id):
+    """Updates the opportunities models with the changes made in the
+    exploration.
+
+    Args:
+        exp_id: str. The exploration id which is also the id of the opportunity
+            model.
+    """
+    updated_exploration = exp_fetchers.get_exploration_by_id(exp_id)
+    content_count = updated_exploration.get_content_count()
+    translation_counts = updated_exploration.get_translation_counts()
+    complete_translation_language_list = (
+        updated_exploration.get_languages_with_complete_translation())
+    model = opportunity_models.ExplorationOpportunitySummaryModel.get(exp_id)
+    exploration_opportunity_summary = (
+        get_exploration_opportunity_summary_from_model(model))
+    exploration_opportunity_summary.content_count = content_count
+    exploration_opportunity_summary.translation_counts = translation_counts
+    exploration_opportunity_summary.complete_translation_languages = (
+        complete_translation_language_list)
+
+    new_languages_for_voiceover = set(complete_translation_language_list) - set(
+        exploration_opportunity_summary.assigned_voice_artist_in_languages)
+
+    # We only append new languages to need_voice_artist_in_languages(instead of
+    # adding all of the complete_translation_language_list), as the
+    # complete translation languages list will be dynamic based on some
+    # content text are changed, where as the voiceover is a long term work and
+    # we can allow a voice_artist to work for an exploration which needs a
+    # little bit update in text translation.
+    need_voice_artist_in_languages_set = set(
+        exploration_opportunity_summary.need_voice_artist_in_languages)
+    need_voice_artist_in_languages_set |= set(new_languages_for_voiceover)
+
+    exploration_opportunity_summary.need_voice_artist_in_languages = list(
+        need_voice_artist_in_languages_set)
+
+    exploration_opportunity_summary.validate()
+
+    _save_multi_exploration_opportunity_summary(
+        [exploration_opportunity_summary])
+
+
+def update_exploration_opportunities_with_new_story(story, exp_ids):
```

Comment:
Prefer passing story_id rather than story object.

Also, can you explain this a little bit -- what is the flow that corresponds to this? I think we delete explorations from a story, and then add them to another story -- I don't think we can just move them from one story to another. /cc @aks681

Reply:
Yes, I understand, but when does this happen? What is the sequence of actions that causes this function to be called -- is it just when an exploration is added to a story?

But before it is added, was there even an opportunity model existing?

Reply:
Oh -- in that case, use ..._with_story_changes rather than ..._with_new_story. The current name makes it sound like it's a different story ID altogether.